htmresearch/frameworks/nlp/classification_model.py

```diff
@@ -371,3 +383,103 @@ def classifyText(self, token, reset=0):
                            implies no decision could be made.
     """
     raise NotImplementedError
+
+
+  def save(self, saveModelDir):
+    """ Save the model in the given directory via pickling.
+
+    @param saveModelDir (string)
+           Directory path for saving the model. This directory should
+           only be used to store a saved model. If the directory does not exist,
+           it will be created automatically and populated with model data. A
+           pre-existing directory will only be accepted if it contains previously
+           saved model data. If such a directory is given, the full contents of
+           the directory will be deleted and replaced with current model data.
+
+    Implementation note: Subclasses should override _serializeExtraData() to
+    save any data that cannot be pickled.
+    """
+    saveModelDir = os.path.abspath(saveModelDir)
+    modelPickleFilePath = os.path.join(saveModelDir, "model.pkl")
+
+    # Delete old model directory if we detect it
+    if os.path.exists(saveModelDir):
+      if not os.path.isdir(saveModelDir) or \
+         not os.path.isfile(modelPickleFilePath):
```

Comment:
Indent here is odd, and it would be better to use implicit continuation w/ parenthesis.